### ISSUE TITLE
Metals assumes that sbt has been run once

### ISSUE THREAD

**Issue by markehammons** [scalameta/metals#3050]
**Describe the bug**

When running metals in a brand new environment that sbt hasn't touched, it fails to connect to bloop because it cannot find certain scala jars.

These are the specific files:

2021.08.18 11:32:29 WARN  Scala instance jar /root/.sbt/boot/scala-2.12.14/lib/scala-library.jar doesn't exist!
2021.08.18 11:32:29 WARN  Scala instance jar /root/.sbt/boot/scala-2.12.14/lib/scala-compiler.jar doesn't exist!
2021.08.18 11:32:29 WARN  Scala instance jar /root/.sbt/boot/scala-2.12.14/lib/scala-reflect.jar doesn't exist!
2021.08.18 11:32:29 WARN  Scala instance jar /root/.sbt/boot/scala-2.12.14/lib/scala-xml_2.12-1.0.6.jar doesn't exist!

**To Reproduce**

1. Add the following Dockerfile to a mill project folder's root:
 ```dockerfile
FROM openjdk:16-jdk-slim

RUN apt-get update && apt-get -y install curl
ADD https://git.io/coursier-cli-linux /usr/local/bin/cs
RUN chmod +x /usr/local/bin/cs
ADD https://github.com/com-lihaoyi/mill/releases/download/0.9.9/0.9.9 /usr/local/bin/mill
RUN chmod +x /usr/local/bin/mill
ENV COURSIER_INSTALL_DIR /usr/local/bin/
RUN cs install sbt
RUN cs install bloop
#RUN sbt -Dsbt.rootdir=true compile
```
2. Using vs remote extension, load project inside of the container
![Screenshot_20210818_132645](https://user-images.githubusercontent.com/2609471/129890218-970ad69b-a402-405e-9efa-c90fb8cca2d7.png)
3. Import mill project in metals

**Expected behavior**
Metals connects to the bloop server

**Screenshots**

![Screenshot_20210818_133450](https://user-images.githubusercontent.com/2609471/129891182-40f40ccb-ff6b-4658-9313-68e97318c913.png)
**Installation:**

- Operating system: Linux container
- Editor: Visual Studio Code/Atom/Vim/Sublime/Emacs
- Metals version: v0.10.5

**Additional context**

This is very certainly caused by the lack of a .sbt folder. Running sbt compile on a non-project folder is enough to fix this issue permanently. The commented out part of my dockerfile fixes the resulting container.
**Search terms**

**Comment by tgodzik** [comment]
Thanks for reporting! This is most likely caused by the `.bloop`  and `.metals` directories being copied to the new container. It contains the configuration that is specific to the current workspace and will not work in another environment. There is really no way around it aside from rerunning build import, since no dependencies will be available in the container. 

Some workarounds that I can think of to deal with it:

- add the Bloop plugin manually and run `sbt -Dbloop.export-jar-classifiers=sources bloopInstall` (more details https://scalameta.org/metals/docs/build-tools/sbt#manual-installation) <- this could be run as a step in the dockerfile.
- remove `.metals` and `.bloop` directories in the dockerfile and you will be prompted to reimport
- try to setup everything in the workspace itself - take a look at the scripts in https://github.com/scalameta/metals-gitpod-sample

Since there is no way Metals can know that we are in a new workspace and that the jars no longer exists, there is no easy way to solve this automatically. 

TLDR; `.bloop` and `.metals` should not be copied to the new workspace.

**Comment by markehammons** [comment]
Reimporting does not work. It's one of the first things I do when I've rebuilt the container. Likewise, I just tried wiping .bloop and .metals right now, and I get the same result after reimporting the build.

The only thing that's been proven to work is running sbt in the fresh container. Once it's been done once, metals acts as expected.

The sbt command you listed does fix the issue, but that's because any invocation of sbt fixes the issue. The workaround I found was to add `RUN sbt -Dsbt.rootdir=true version` to the container definition. It sucks that I have to resort to this though, shouldn't need to pull in sbt at all.

**Comment by tgodzik** [comment]
> Reimporting does not work.

Reimporting does actually run `sbt -Dbloop.export-jar-classifiers=sources bloopInstall`, so that should work exactly the same as running the command manually. Did you remove .bloop and .metals directories in the Docker file? In that case we treat the container as a new workspace and everything should work correctly.

The problem with the workflow here is that the build tool is the one that downloads the dependencies and puts them into configuration that Bloop and Metals use. If that disappears we need to rerun the build tool to make it download all the jars again.

**Comment by tgodzik** [comment]
Also, all the paths might be broken in the container, so there is no way rather than reimporting the project once again. I would imagine VS Code would need to provide some kind of a virtual file system that we would need to make it work.
